lib/trino-parquet/src/main/java/io/trino/parquet/metadata/ParquetMetadata.java

```diff
@@ -153,6 +151,29 @@ public List<BlockMetadata> getBlocks(long splitStart, long splitLength)
         return blocks;
     }
 
+    // check the rowGroup whether overlap with the required split range
+    private static boolean notOverLap(long splitStart, long splitLength, RowGroup rowGroup)
+    {
+        long rowGroupStart = rowGroup.getFile_offset();
+        long rowGroupEnd = rowGroupStart + rowGroup.getTotal_compressed_size();
+
+        // only check the splitStart is not enough as there is case that
+        // the rowGroupEnd fit within split range
+
+        // check rowGroupEnd locate within the split range
+        if (rowGroupEnd < splitStart) {
+            return true;
+        }
+
+        // check rowGroupStart locate within the split range
+        // avoid potentially overflow
```

Comment:
With long values it would take a several exabytes size file to hit overflow, which would make it an impractical situation. We should rather focus on clarity of the logic here.

Reply:
I notice there exists already a method `getBlocks(0, Long.MAX_VALUE)`, that's why I have this concern

Reply:
That's a helper method to simplify usage when we want to read all blocks. Does that cause overflow anywhere ?

Reply:
I haven't found, just thought the `splitLength` could be large